rls/src/main/java/io/grpc/rls/CachingRlsLbClient.java

```diff
@@ -563,9 +573,29 @@ void maybeRefresh() {
       }
     }
 
+    @VisibleForTesting
+    ChildPolicyWrapper getChildPolicyWrapper(String target) {
+      for (ChildPolicyWrapper childPolicyWrapper : childPolicyWrappers) {
+        if (childPolicyWrapper.getTarget().equals(target)) {
+          return childPolicyWrapper;
+        }
+      }
+
+      throw new RuntimeException("Target not found:" + target);
+    }
+
     @Nullable
     ChildPolicyWrapper getChildPolicyWrapper() {
-      return childPolicyWrapper;
+      if (childPolicyWrappers == null || childPolicyWrappers.isEmpty()) {
```

Comment:
How could the wrappers be null?

We should validate that there are targets when constructing the DataCacheEntry. It looks like that is safe to do in the constructor as all calls to `new DataCacheEntry()` are in a try-catch. That would avoid passing `null` to the data plane.

Reply:
You are correct that wrappers can't be null and resp.targets() can't be null, but if the response target list is empty, then the wrappers would be also.  Is it legal for RLS to return an empty list of targets?

Reply:
I don't think it matters. If we don't have a list of targets, whether it is allowed or not, either way we're going to fail the RPC.